### PR TITLE
[Misc]Add customized information for models

### PR DESCRIPTION
**Pull request by jeejeelee**
When I debug the VLLM, the model's print output always bothers me because it lacks details, as shown below:
```python
LlavaForConditionalGeneration(
  (vision_tower): CLIPVisionModel(
    (vision_model): CLIPVisionTransformer(
      (embeddings): CLIPVisionEmbeddings(
        (patch_embedding): Conv2d(3, 1024, kernel_size=(14, 14), stride=(14, 14), bias=False)
        (position_embedding): Embedding(577, 1024)
      )
      (pre_layrnorm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
      (encoder): CLIPEncoder(
        (layers): ModuleList(
          (0-23): 24 x CLIPEncoderLayer(
            (self_attn): CLIPAttention(
              (k_proj): Linear(in_features=1024, out_features=1024, bias=True)
              (v_proj): Linear(in_features=1024, out_features=1024, bias=True)
              (q_proj): Linear(in_features=1024, out_features=1024, bias=True)
              (out_proj): Linear(in_features=1024, out_features=1024, bias=True)
            )
            (layer_norm1): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
            (mlp): CLIPMLP(
              (activation_fn): QuickGELUActivation()
              (fc1): Linear(in_features=1024, out_features=4096, bias=True)
              (fc2): Linear(in_features=4096, out_features=1024, bias=True)
            )
            (layer_norm2): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
          )
        )
      )
      (post_layernorm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
    )
  )
  (multi_modal_projector): LlavaMultiModalProjector(
    (linear_1): Linear(in_features=1024, out_features=4096, bias=True)
    (act): GELU(approximate='none')
    (linear_2): Linear(in_features=4096, out_features=4096, bias=True)
  )
  (language_model): LlamaModel(
    (embed_tokens): VocabParallelEmbedding()
    (layers): ModuleList(
      (0-31): 32 x LlamaDecoderLayer(
        (self_attn): LlamaAttention(
          (qkv_proj): QKVParallelLinear()
          (o_proj): RowParallelLinear()
          (rotary_emb): RotaryEmbedding()
          (attn): Attention()
        )
        (mlp): LlamaMLP(
          (gate_up_proj): MergedColumnParallelLinear()
          (down_proj): RowParallelLinear()
          (act_fn): SiluAndMul()
        )
        (input_layernorm): RMSNorm()
        (post_attention_layernorm): RMSNorm()
      )
    )
    (norm): RMSNorm()
  )
  (lm_head): ParallelLMHead()
  (logits_processor): LogitsProcessor()
  (sampler): Sampler()
)
```

By leveraging [extra_repr](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.extra_repr), we can add more details to the model, and can achieve the following print output:

```python
LlavaForConditionalGeneration(
  (vision_tower): CLIPVisionModel(
    (vision_model): CLIPVisionTransformer(
      (embeddings): CLIPVisionEmbeddings(
        (patch_embedding): Conv2d(3, 1024, kernel_size=(14, 14), stride=(14, 14), bias=False)
        (position_embedding): Embedding(577, 1024)
      )
      (pre_layrnorm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
      (encoder): CLIPEncoder(
        (layers): ModuleList(
          (0-23): 24 x CLIPEncoderLayer(
            (self_attn): CLIPAttention(
              (k_proj): Linear(in_features=1024, out_features=1024, bias=True)
              (v_proj): Linear(in_features=1024, out_features=1024, bias=True)
              (q_proj): Linear(in_features=1024, out_features=1024, bias=True)
              (out_proj): Linear(in_features=1024, out_features=1024, bias=True)
            )
            (layer_norm1): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
            (mlp): CLIPMLP(
              (activation_fn): QuickGELUActivation()
              (fc1): Linear(in_features=1024, out_features=4096, bias=True)
              (fc2): Linear(in_features=4096, out_features=1024, bias=True)
            )
            (layer_norm2): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
          )
        )
      )
      (post_layernorm): LayerNorm((1024,), eps=1e-05, elementwise_affine=True)
    )
  )
  (multi_modal_projector): LlavaMultiModalProjector(
    (linear_1): Linear(in_features=1024, out_features=4096, bias=True)
    (act): GELU(approximate='none')
    (linear_2): Linear(in_features=4096, out_features=4096, bias=True)
  )
  (language_model): LlamaModel(
    (embed_tokens): VocabParallelEmbedding(num_embeddings=32064, embedding_dim=4096, org_vocab_size=32064, num_embeddings_padded=32064, tp_size=1)
    (layers): ModuleList(
      (0-31): 32 x LlamaDecoderLayer(
        (self_attn): LlamaAttention(
          (qkv_proj): QKVParallelLinear(in_features=4096, output_features=12288, bias=False, tp_size=1, gather_output=False)
          (o_proj): RowParallelLinear(input_features=4096, output_features=4096, bias=False, tp_size=1, reduce_results=True)
          (rotary_emb): RotaryEmbedding(head_size=128, rotary_dim=128, max_position_embeddings=4096, base=10000.0, is_neox_style=True)
          (attn): Attention(head_size=128, num_heads=32, num_kv_heads=32, scale=0.08838834764831845)
        )
        (mlp): LlamaMLP(
          (gate_up_proj): MergedColumnParallelLinear(in_features=4096, output_features=22016, bias=False, tp_size=1, gather_output=False)
          (down_proj): RowParallelLinear(input_features=11008, output_features=4096, bias=False, tp_size=1, reduce_results=True)
          (act_fn): SiluAndMul()
        )
        (input_layernorm): RMSNorm(hidden_size=4096, eps=1e-05)
        (post_attention_layernorm): RMSNorm(hidden_size=4096, eps=1e-05)
      )
    )
    (norm): RMSNorm(hidden_size=4096, eps=1e-05)
  )
  (lm_head): ParallelLMHead(num_embeddings=32064, embedding_dim=4096, org_vocab_size=32064, num_embeddings_padded=32064, tp_size=1)
  (logits_processor): LogitsProcessor(vocab_size=32064, org_vocab_size=32064, scale=1.0, logits_as_input=False)
  (sampler): Sampler()
```

### CODE REVIEW

**Comment by jeejeelee** [comment]
@zhuohan123  could I trouble you to review this PR, I am not sure whether this feature is useful. If not, I will close this PR.

File: vllm/model_executor/layers/logits_processor.py [hunk 1]
```diff
@@ -70,6 +70,11 @@ def _get_logits(self, hidden_states: torch.Tensor, embedding: torch.Tensor,
             logits = logits[:, :self.org_vocab_size]
         return logits
 
+    def extra_repr(self) -> str:
+        s = "vocab_size={vocab_size}, org_vocab_size={org_vocab_size}"
+        s += ", scale={scale}, logits_as_input={logits_as_input}"
+        return s.format(**self.__dict__)
```
Comment (mgoin) [inline vllm/model_executor/layers/logits_processor.py:76]: why not use an f-string here?

File: vllm/model_executor/layers/rotary_embedding.py [hunk 2]
```diff
@@ -155,6 +155,12 @@ def forward(
                                  self.cos_sin_cache, self.is_neox_style)
         return query, key
 
+    def extra_repr(self) -> str:
+        s = "head_size={head_size}, rotary_dim={rotary_dim}"
+        s += ", max_position_embeddings={max_position_embeddings}"
+        s += ", base={base}, is_neox_style={is_neox_style}"
+        return s.format(**self.__dict__)
```
Comment (mgoin) [inline vllm/model_executor/layers/rotary_embedding.py:162]: ditto - why not use an f-string here?

**Review by mgoin — COMMENTED** [review]
I think this would be pretty useful!

File: vllm/model_executor/layers/logits_processor.py (hunk 1 above)
Comment (jeejeelee) [reply vllm/model_executor/layers/logits_processor.py:76]: I referred to the code from  https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/sparse.py#L167. Indeed, I should ensure consistent use of f-strings :smile: , I will process this as soon as possible

File: vllm/model_executor/layers/rotary_embedding.py (hunk 2 above)
Comment (jeejeelee) [reply vllm/model_executor/layers/rotary_embedding.py:162]: ditto

**Review by mgoin — APPROVED** [review]
lgtm, thank you

**Comment by jeejeelee** [comment]
@mgoin  It seems  that we also need a collaborator to review this PR, is that right?

**Comment by jeejeelee** [comment]
@simon-mo  could I trouble you to take a look at this PR, please?

**Comment by jeejeelee** [comment]
@zhuohan123  @simon-mo  Could you please take a look at this PR? If it's not useful, I'll close it. I'm hoping to get some feedback. :bowtie:

**Comment by youkaichao** [comment]
I think this is a good direction in general. You should add comment to point to the doc https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.extra_repr , this feature is not well-known.

In addition, please merge main into this branch to pass the CI.

**Pull request merged**